pkg/beholder/logger/logger.go

```diff
@@ -0,0 +1,66 @@
+package logger
+
+import (
+	"log"
+	"os"
+
+	"github.com/uptrace/opentelemetry-go-extra/otelzap"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// TODO(gg): use or remove this package
+
+//go:generate mockery --name Logger --output ./mocks/ --case=underscore
+type Logger interface {
+	Info(args ...interface{})
+	Infow(msg string, keysAndValues ...interface{})
+	Infof(format string, values ...interface{})
+
+	Debug(args ...interface{})
+	Debugw(msg string, keysAndValues ...interface{})
+	Debugf(format string, values ...interface{})
+
+	Warn(args ...interface{})
+	Warnw(msg string, keysAndValues ...interface{})
+	Warnf(format string, values ...interface{})
+
+	Error(args ...interface{})
+	Errorw(msg string, keysAndValues ...interface{})
+	Errorf(format string, values ...interface{})
+
+	Panic(args ...interface{})
+	Panicf(format string, values ...interface{})
+
+	Fatal(args ...interface{})
+	Fatalf(format string, values ...interface{})
+
+	Sync() error
+}
+
+func New() Logger {
+	return newZapLogger().Sugar()
+}
+
+// NewOtelzapLogger records Zap log messages as messages on the existing span that must be passed in a context.Context as a first argument.
+// It does not record anything if the context does not contain a span.
+// See more details [here](https://github.com/uptrace/opentelemetry-go-extra/tree/main/otelzap)
+func NewOtelzapLogger() *otelzap.Logger {
+	zl := newZapLogger()
+	return otelzap.New(newZapLogger(), otelzap.WithMinLevel(zl.Level()))
```

Comment:
I don't think this approach will work well with our existing loggers. We would have to switch over, rather than just hooking in and forwarding from the existing API. Is there a way to get an otel zap core that we just plug in to our existing logger?

Reply:
https://pkg.go.dev/github.com/agoda-com/opentelemetry-go/otelzap#NewOtelCore

Reply:
@jmank88 
https://pkg.go.dev/github.com/agoda-com/opentelemetry-go/otelzap
package depends on [github.com/agoda-com/opentelemetry-logs-go](https://github.com/agoda-com/opentelemetry-logs-go) which is a custom implementation of https://github.com/open-telemetry/opentelemetry-go Logs [sdk](https://github.com/open-telemetry/opentelemetry-go/tree/main/sdk/log).
Because of that it's incompatible with current Beholder implementation which relies on open-telemetry/opentelemetry-go [sdk](https://github.com/open-telemetry/opentelemetry-go/tree/main/sdk/log).  

I'm going to remove beholder/logger based on otelzap from the scope of this PR.